src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,535 @@
+
+```astro title="src/pages/posts/[slug].astro" ins={3,11-19}
+---
+import { contentfulClient } from "../../lib/contentful";
+import { documentToHtmlString } from "@contentful/rich-text-html-renderer";
+import type { blogPostFields } from "../../lib/contentful";
+
+export async function getStaticPaths() {
+  const entries = await contentfulClient.getEntries<blogPostFields>({
+    content_type: "blogPost",
+  });
+
+  const pages = entries.items.map((item) => ({
+    params: { slug: item.fields.slug },
+    props: {
+      title: item.fields.title,
+      content: documentToHtmlString(item.fields.content),
+      date: new Date(item.fields.date).toLocaleDateString(),
+    },
+  }));
+  return pages;
+}
+---
+```
+
+In this example, the `params` property is an object with a `slug` property. The property inside `params` should match the name of the dynamic route (i.e `[slug].astro`).
+
+The `props` property is an object with the properties: title, content and date. We will use the `documentToHtmlString` method to convert the `content` property from a Document to a string of HTML and the Date constructor to format the date.
+
+Finally, you can use the page `props` to display your blog post.
+
+```astro title="src/pages/posts/[slug].astro" ins={21,23-32}
+---
+import { contentfulClient } from "../../lib/contentful";
+import { documentToHtmlString } from "@contentful/rich-text-html-renderer";
+import type { blogPost } from "../../lib/contentful";
+
+export async function getStaticPaths() {
+  const { items } = await contentfulClient.getEntries<blogPost>({
+    content_type: "blogPost",
+  });
+  const pages = items.map((item) => ({
+    params: { slug: item.fields.slug },
+    props: {
+      title: item.fields.title,
+      content: documentToHtmlString(item.fields.content),
+      date: new Date(item.fields.date).toLocaleDateString(),
+    },
+  }));
+  return pages;
+}
+
+const { content, title, date } = Astro.props;
+---
+<html lang="en">
+  <head>
+    <title>{title}</title>
+  </head>
+  <body>
+    <h1>{title}</h1>
+    <time>{date}</time>
+    <article set:html={content} />
+  </body>
+</html>
+```
+
+Navigate to http://localhost:3000/posts/astro-is-amazing/ to make sure your dynamic route is working!
+
+#### Server side rendering
+
+To render a blog post on the server, you will use the frontmatter to request the data from Contentful and pass it to the page as props.
+
+Import the `blogPost` interface and `contentfulClient` from `src/lib/contentful.ts`. Use the dynamic route parameter `slug` and the `getEntries` method from `contentfulClient` to fetch the desired entry with a content type of `blogPost` and a slug that matches the dynamic route parameter.
+
+You can use the [`Astro.params`](/en/core-concepts/routing/#the-astroparams-object) object to get the parameters from the URL. 
```

Comment:
This is also hard to follow without the code; I think we can jump straight to it, especially since we explained the setup in the section before the tutorial:

```suggestion
Create a `[slug].astro` page in `src/pages/posts`. Use [`Astro.params`](/en/core-concepts/routing/#the-astroparams-object) to get the slug from the URL, then pass that to `getEntries`: 
```